runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -777,6 +818,17 @@ ClassFileOracle::walkMethods()
 		if (methodIsObjectConstructor(methodIndex)) {
 			_methodsInfo[methodIndex].modifiers |= J9AccMethodObjectConstructor;
 		}
+		
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		if (methodIsConstructor(methodIndex)) {
```

Comment:
Would it be better to check `if (!_hasNonEmptyConstructor) {` before `if (methodIsConstructor(methodIndex)) {`?

This way if you have already seen a constructor you can just skip everything.

Reply:
also, in `ClassFileOracle::ClassFileOracle(..)` if the class is a VT or a concrete class you can just set `_hasNonEmptyConstructor = true`.

Reply:
> Would it be better to check if (!_hasNonEmptyConstructor) { before if (methodIsConstructor(methodIndex)) {?

Yes.

Reply:
Fixed.